src/index.ts

```diff
@@ -53,4 +53,13 @@ const useProxy = <T extends object>(p: T): T => {
   return createDeepProxy(currSnapshot, affected, proxyCache)
 }
 
-export { proxy, subscribe, snapshot, useProxy }
+const useLocalProxy = <T extends object>(init: T | (() => T)) => {
+  const ref = useRef<T>()
+  if (!ref.current) {
+    const initialObject = typeof init === 'function' ? (init as any)() : init
```

Comment:
Just for the suggestion, Can we use `(init as () => T)` instead of `(init as any)`?

Reply:
ah yes ofc - I think @dai-shi is going to take over this PR but went ahead and threw that update in bc I agree `any` is ugly